fix(UI): enhancements to the admin dashboard page issue #532

NewtonLC opened the pull request:
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a [descriptive title](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request?id=prepare-a-good-pr-title) (**not** a vague title like `Update index.md`)

Closes #532

<!-- Feel free to add any additional description of changes below this line -->
Worked in collaboration with @CarlyAThomas

In relation to [post](https://github.com/freeCodeCamp/classroom/issues/532#issuecomment-2606056975).

GuillermoFloresV commented on components/adminTable.js:
```diff
@@ -99,6 +123,94 @@ export default function AdminTable(props) {
+                  <select
+                    value={entriesPerPage}
+                    onChange={e => setEntriesPerPage(Number(e.target.value))}
+                  >
+                    <option value={10}>10</option>
+                    <option value={20}>20</option>
+                    <option value={30}>30</option>
+                    <option value={40}>40</option>
+                    <option value={50}>50</option>
+                  </select>
```
is there a reason why this is hardcoded to 50?

GuillermoFloresV commented on components/adminTable.js:
```diff
@@ -99,6 +123,94 @@ export default function AdminTable(props) {
+                >
+                  {startEntry}-{endEntry} of {totalEntries}
+                </span>
+                <button
+                  onClick={() => setPageIndex(0)}
+                  disabled={!canPreviousPage}
+                  style={{
+                    marginLeft: '10px',
+                    marginRight: '10px',
+                    color: '#d1d1d1',
+                    fontSize: '1.5em'
+                  }}
+                >
+                  |<b>&lt;</b>
```
I see a lot of repeated styling code in the buttons, would we be able to cut down on this?

GuillermoFloresV commented on components/adminTable.js:
```diff
@@ -44,12 +47,24 @@ export default function AdminTable(props) {
+
+  const startEntry = pageIndex * entriesPerPage + 1;
+  const endEntry = Math.min((pageIndex + 1) * entriesPerPage, rows.length);
+  const totalEntries = rows.length;
```
We should leverage this variable for the pagination feature, we should be able to dynamically create the options for the footer, removing the hardcoded options. We can just generate some mock data to fill in the students in a class so that we can simulate the footer still displaying and working correctly?  Let me know your thoughts on this.

utsab commented on components/adminTable.js:
```diff
@@ -44,12 +47,42 @@ export default function AdminTable(props) {
+    marginRight: '10px',
+    color: disabled ? '#d1d1d1' : '#757575',
+    fontSize: '1.5em'
+  });
```
When declaring a css style rule that you wish to reuse, a better practice is to create a separate css file for the module.

NewtonLC commented:
Hi @utsab, @CarlyAThomas and I have moved the styling from being hard-coded in `adminTable.js` to a separate file, `adminTable.module.css`. We tested it manually and haven't noticed any changes in styling or behavior. We also updated the adminTable snapshot. Can you confirm that our changes are correct?
